pos-print/src/main/java/com/clouway/pos/print/core/PrintQueue.kt

```diff
@@ -0,0 +1,31 @@
+package com.clouway.pos.print.core
+
+/**
+ * Provides the methods to iterate over and append to a
+ * queue of receipts.
+ *
+ * @author Tsvetozar Bonev (tsvetozar.bonev@clouway.com)
+ */
+interface PrintQueue {
+
+  /**
+   * Checks if there is a next receipt, blocks until a next
+   * one is found.
+   */
+  fun hasNext(): Boolean
```

Comment:
I'm not sure that you need hasNext. next() could block and return nullable value if no items are available form processing.

For example take a look into: https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/ArrayBlockingQueue.html#poll--

Reply:
Implemented blocking via the ArrayBlockingQueue take() method.